executor/prepared_test.go

```diff
@@ -157,7 +158,7 @@ func (sm *mockSessionManager2) GetProcessInfo(id uint64) (pi *util.ProcessInfo,
 	return
 }
 func (sm *mockSessionManager2) Kill(connectionID uint64, query bool) {
-	sm.killed = true
```

Comment:
I suspect the root cause is that the kill command is send more than one time.
Because if the `killed` is not set, the query will not exit (assume the test logic is correct).
So set `killed` = true happens before the query execution.
and query execution happens before check `killed` flag, (obviously they are in the same goroutine)
... that mean line 196 check `killed` flag should happens before the set `killed` operation...

From the above deduction, checking flag and setting flag operation have the happens before relationship, they happen in different time and should data race.

The kill command execute twice is more likely to be the root reason.

Reply:
Assume the `Kill()` is executed more than once, change to channel may cause the kill operation block, and goroutine leak in the test.

How about changing to the atomic way?

Reply:
I see, it is indeed possible for Kill to be executed more than once, I changed it to use atomic.